### ISSUE TITLE
City saving bug

### ISSUE THREAD

**Issue by Rushing** [Deceth/Battle-City#20]
For some reason, when I create, for example, a large Gauntlet, when I save it, some building sections aren't added. It's not a sector issue, because parts of the city that where two or three sectors away were still built.
The only thing plausible is that those sections were off the (M) button On Screen Map

**Comment by Deceth** [comment]
It is a sector issue. When you play you will have several sectors loaded at a time. When you save a city, it will only be able to save sectors that are currently loaded. 

Opening the map before you save is a good way to know what will actually be saved.

**Comment by Rushing** [comment]
Darnit, is it not possible to simply check what sectors contain your buildings, and load them temporarily for the saving process?

**Comment by Deceth** [comment]
Of course it is. It just doesn't do that.

**Comment by Rushing** [comment]
Why not! If there is no plausible reason for it NOT doing it, then surely it's just dumb for it not to.

**Comment by Deceth** [comment]
Are you trying to philosophically dispute bugs out of existence?

**Comment by Rushing** [comment]
Some bugs there are logical and reasonably explanations for why they can't be fixed, for example, if loading all these sectors caused an incredible amount of lag, then I would understand not fixing it.

**Comment by Deceth** [comment]
You appear to be in a different conversation. In this one, you mentioned a bug, I explained the issue, you proposed a solution, I said yes that's possible =p

**Comment by Rushing** [comment]
And after you said it's possible you said that it simply doesn't, to which I replied why not, and gave a simple point towards my example. Then we both went off topic ;)

**Comment by Deceth** [comment]
Yes, you said 'why not'. That's like asking your grandfather, why didn't you buy Microsoft stock 25 years ago?! 

Unless you have a crystal ball to see into the future, then the obvious answer is nobody foresaw this bug, and therefore nobody foresaw the alternative solution which only exists because of the bug...

**Comment by Rushing** [comment]
I have been outsmarted :(
But that still leaves me wondering, are you going to fix it? And whilst on the subject of you doing things, are you going to get around to putting back static cities?
If the answer is no to either of these, then I request you help me with getting Fmod and Directx set up properly so I can do it myself :o

**Comment by Deceth** [comment]
Wait, who said anything about me doing things? The only thing I've committed to doing is plotting the death of the Battle City Server if my demands are not met. I'm not a programmer, I'm an Evil Overlord... And I'm certainly not fmod or DirectX support, look up some documentation, I have no clue how to set that up.

If you need help building a moat, transforming into a giant serpent or ranting about the inconvenience of shapes like the rhombus, then maybe I can help.

**Comment by Rushing** [comment]
I do need a new moat...

**Comment by Deceth** [comment]
Never buy the gators first :(

**Comment by Deceth** [comment]
City saving works for cities of reasonable size, but it won't work if the mayor is far from this city. As has sort of been discussed here, it looks like the sectors that are loaded are the areas you can see when you open up the "map". 

So if the mayor is standing in a spot far enough from the city where half his/her city is not loaded at the moment, then only the parts loaded will save.

So, when saving a city, rather than use the Mayor's location, the game should first load the sectors around the Mayors Command Center, then save to ensure this doesn't occur.

Saving Admin Built Giant Gauntlet cities, that should be a separate issue. It's not a common use case we should address with this bug, it's a much lower priority unless it's feasible to load up all the sectors containing a cities buildings in an easy enough way.
